### ISSUE TITLE
Layout/DotPosition with EnforcedStyle of 'trailing' is confused by heredocs with `.squish`

### ISSUE THREAD

**Issue by codebender** [rubocop/rubocop#10144]
I ran into this after upgrading rubocop from version `1.17.0` to `1.22.0`. 

After adding `.squish` to a heredoc, which rails rubocop suggested, Layout/DotPosition started reporting `Place the . on the previous line, together with the method call receiver.`.  The problem does seem related to the `EnforcedStyle: trailing` for that cop. 

**Example:**
```ruby
class Test
  def query
    <<-SQL.squish
      SELECT * FROM tests
    SQL
  end
end
```

This [recent PR](https://github.com/rubocop/rubocop/pull/10113) added a different fix for Layout/DotPosition & heredocs, but might have introduced this bug. 🤷‍♂️

--------

## Expected behavior
No reported problems with the code

## Actual behavior

Describe here what actually happened.
`Layout/DotPosition: Place the . on the previous line, together with the method call receiver. <<-SQL.squish`

## Steps to reproduce the problem

Add `.squish` to a heredoc

## RuboCop version

```
$ bundle exec rubocop -V                                                                                                                                          
1.22.0 (using Parser 3.0.2.0, rubocop-ast 1.12.0, running on ruby 2.7.4 x86_64-darwin19)
  - rubocop-performance 1.11.5
  - rubocop-rails 2.12.2
```

**Comment by dvandersluis** [comment]
Thanks, but this is actually a duplicate of #10140, which is already fixed (but not merged) by #10141.

**Comment by codebender** [comment]
Oh sorry about that, I didn't see that one 👍

**Comment by dvandersluis** [comment]
No worries - I’ve updated the title of the other ticket to hopefully be easier to find now! Thanks again!
